Automattic/jetpack
# Add: Jetpack App as a new upload media extension

This PR adds a new modal that is found in the Select image drop down for a bunch of image/gallery blocks. 
See

Clicking the "Jetpack Mobile App" button you end up with 
<img width="629" alt="Screenshot 2024-01-10 at 1 15 46 PM" src="https://github.com/Automattic/jetpack/assets/115071/89393caf-6390-4178-bc5a-a6787d8b2d5c">

<img width="742" alt="Screenshot 2024-01-10 at 1 16 03 PM" src="https://github.com/Automattic/jetpack/assets/115071/315c4dde-ec69-4424-8b64-7e2095d449e9">




And once you upload media via the Jetpack app. (or in a new tab) 
You can then see the newly added media you see something like 

<img width="754" alt="Screenshot 2024-01-10 at 1 16 29 PM" src="https://github.com/Automattic/jetpack/assets/115071/12d2f533-c5df-49f6-926c-152ba313ca4b">

The image then gets added once the photo is selected. 

Here is a video. 


https://github.com/Automattic/jetpack/assets/115071/ab3ab677-fb2c-466d-bc6f-231c2c06e7c0


## Testing instructions:
 Set `define( 'JETPACK_EXPERIMENTAL_BLOCKS', true ); `

Visit the editor. Notice the new Jetpack App Media modal when you try to add/replace image in a block. Sucha s image, gallery, tile gallery block. 

You can use the beta version for the apps running Jetpack App version 24.0 or later. (not yet released) 
Or you can open up a new tab and upload media under the same user. You should see the modal change and let you enter see the newly uploaded media. 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to '..'
